### DISCUSSION TITLE
Multi-Line Dialogue Issue

### DISCUSSION THREAD

**Discussion by iaptekarev33** [nathanhoad/godot_dialogue_manager#174] [General]
**Hi Nathan, here is the code copied from my youtube comment regarding displaying multiple lines of code. I can provide more info if you need it. My guess is I am misunderstanding how unhandled input is used, but maybe there is a better method to do this? Thank you so much for your help!**

I am having some trouble with implementing multiple lines of dialogue in Godot 3.5.
 
My dialogue looks like this:
~ spider
Spider: I am a spider
Spider: Second Line

The player interacts with the NPC like this ("key" is an attribute that the NPCs have which corresponds to a dialogue title):

    func _unhandled_input(event):
        if event.is_action_pressed("ui_accept"):
            if ray.is_colliding():
                var key = ray.get_collider().dialogue_key
                emit_signal("player_dialogue", key)

The player dialogue signal initiates this function in the game scene which should create the dialogue balloon. This works for the first line of dialogue, but for the second line the balloon comes up but there is no text

    func _on_Player_player_dialogue(key):

        var line = yield(DialogueManager.get_next_dialogue_line(key, dialogue_resource), "completed")

        if line != null:
            var balloon = preload("res://UI/Dialogue Balloon/dialogue_balloon.tscn").instance()
            balloon.dialogue_line = line
            UI.add_child(balloon)
            DialogueManager.show_example_dialogue_balloon(yield(balloon, "actioned"), dialogue_resource)

Any tips?? I really appreciate your help!!

**Comment by iaptekarev33** [comment]
So I deleted the plugin and re-added it and now the first line looks fine as before, and the second line displays, however it appears as the example balloon and not my custom balloon.

**Reply by nathanhoad** [reply]
The custom balloon has to be made outside of the addon folder otherwise it will get wiped whenever you update the plugin.
